src/util/cacheManager.ts

```diff
@@ -0,0 +1,66 @@
+import { Message } from 'discord.js';
+import { journalCache, gratefulnessCache} from '../data/channelCaches'
+
+/**
+ * @param msg The message object.
+ * @param channel The channel whose cache needs to be updated. 'j' stands for Journal and 'g' for Gratefulness
+ */
+export function addEntry(msg : Message, channel : 'j'|'g') : void{
```

Comment:
Instead of having `'j' | 'g'` in many places, you could refactor this. Better yet, you could create a map from string to journal:

```
const foo = {"journal": journalCache, "gratefulness": gratefulnessCache }
```

And then to get the individual strings, you could use `keyof typeof foo`.